### PR TITLE
Fix a few issues in DeviceConformingProlongationOperator

### PR DESCRIPTION
**Pull request by v-dobrev**
This PR fixes the issues uncovered by tracking down the reason for the diffs reported here: https://github.com/mfem/mfem/pull/1644#issuecomment-806360144.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2148](https://github.com/mfem/mfem/pull/2148) | @v-dobrev | @tzanio | @artv3 + @camierjs | 04/01/21 | 04/04/21 | 04/13/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by artv3 — APPROVED** [review]

**Review by tomstitt — APPROVED** [review]

**Review by camierjs — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
@camierjs, @artv3, @tomstitt, @tzanio,

I pushed a few small tweaks to optimize one potential use case. Can you take another look at this?

**Review by tzanio — APPROVED** [review]

File: fem/pfespace.cpp [hunk 1]
```diff
@@ -3256,61 +3257,68 @@ DeviceConformingProlongationOperator::DeviceConformingProlongationOperator(
...
 void DeviceConformingProlongationOperator::Mult(const Vector &x,
                                                 Vector &y) const
 {
    const GroupTopology &gtopo = gc.GetGroupTopology();
    int req_counter = 0;
+   // Make sure 'y' is marked as valid on device and for use on device.
+   // This ensures that there is no unnecessary host to device copy when the
+   // input 'y' is valid on host (in 'y.SetSubVector(ext_ldof, 0.0)' when local
+   // is true) or BcastLocalCopy (when local is false).
+   y.Write();
```
Comment (artv3) [inline fem/pfespace.cpp:3317]: Nice!

**Review by artv3 — APPROVED** [review]
Excelente!

File: fem/pfespace.cpp [hunk 2]
```diff
@@ -3256,61 +3257,68 @@ DeviceConformingProlongationOperator::DeviceConformingProlongationOperator(
...
 }
 
-static void SetSubVector(const int N,
-                         const Array<int> &indices,
+static void SetSubVector(const Array<int> &indices,
                          const Vector &in, Vector &out)
 {
-   auto y = out.Write();
+   MFEM_ASSERT(indices.Size() == in.Size(), "incompatible sizes!");
+   // Use ReadWrite() since we modify only a sebset of the indices:
```
Comment (camierjs) [inline fem/pfespace.cpp:3285]: Typo

**Review by tomstitt — APPROVED** [review]

File: fem/pfespace.cpp (hunk 2 above)
Comment (v-dobrev) [reply fem/pfespace.cpp]: Fixed.

**Review by camierjs — APPROVED** [review]

**Comment by tzanio** [comment]
Re-merged in `next` for testing ...

**Pull request merged**